### PR TITLE
Add classes

### PR DESCRIPTION
**Pull request by ahmed025566**
In this milestone you could see that I have done the following: 

- I have created the Person class which is the super class.
- I have created Student class which is a sub class (inherited from Person).
-  I have created Teacher class which is a sub class (inherited from Person).

### CODE REVIEW

File: teacher.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require_relative 'person'
+
+class Teacher < Person
+  def initialize(age, name, specialization)
+    super(age, name)
+    @specialization = specialization
+  end
```
Comment (Owusu-Desmond) [inline teacher.rb]: - [ ] Your teacher class constructor is missing the `parent_permission` parameter, which should have a default value of `true`.  Your constructor method should look similar to this :point_down: 
  
  ```ruby
  def initialize(age, specialization, name = "Unknown", parent_permission = true)
      super(age, name, parent_permission)
      @specialization = specialization
    end
  ```

File: student.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+require_relative 'person'
+
+class Student < Person
+  def initialize(age, name, classroom)
+    super(age, name)
```
Comment (Owusu-Desmond) [inline student.rb]: - [ ] Your student class constructor is missing the `parent_permission` parameter, which should have a default value of `true`.  Your constructor method should look similar to this :point_down: 
    ```ruby
  def initialize(age, classroom, name = "Unknown", parent_permission = true)  
      super(age, name, parent_permission)
      @classroom = classroom
    end
  ```

**Review by Owusu-Desmond — CHANGES_REQUESTED** [review]
Hi @ahmed025566 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
📌 No Linters error ✅
📌 Descriptive pull request✅
📌 Good commit messages ✅
📌 Professional README file ✅

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: student.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+require_relative 'person'
+
+class Student < Person
+  def initialize(age, classroom, parent_permission, name)
+    super(age, name, parent_permission)
+    @classroom = classroom
+  end
```
Comment (thecodechaser) [inline student.rb]: - [ ] As requested by the previous reviewer please make sure the `parent_permission` parameter has a default value as `true`.

File: teacher.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+require_relative 'person'
+
+class Teacher < Person
+  def initialize(age, name, parent_permission, specialization)
+    super(age, name, parent_permission)
+    @specialization = specialization
+  end
```
Comment (thecodechaser) [inline teacher.rb]: - [ ] As requested by the previous reviewer please make sure the `parent_permission` parameter has a default value as `true`.

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @ahmed025566👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by AbelG101 — APPROVED** [review]
# Hi @ahmed025566 👋, 

**Your project is complete!  There is nothing else to say other than... it's time to merge it. 
Congratulations! 🎉 🥳🥇**

## Highlights
- Proper class implementations ✔
- Correct constructor initialization & inheritance ✔
- Correct use of Git Flow ✔
- All linter checks are passing ✔
- Descriptive PR title with a great summary ✔

## Optional suggestions
- N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**